File: output/qwen_2.5_72b_instruct/codenet/direct_translation/python/java/s663509070.java
import java.util.Scanner;

public class s663509070 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int D = scanner.nextInt();
        int[][] mat = new int[N][D];
        for (int d = 0; d < N; d++) {
            for (int j = 0; j < D; j++) {
                mat[d][j] = scanner.nextInt();
            }
        }
        int c = 0;
        for (int i = 0; i < mat.length - 1; i++) {
            for (int j = i + 1; j < mat.length; j++) {
                double dist = Math.sqrt(sumOfSquaredDifferences(mat[i], mat[j]));
                if (dist == (int) dist) {
                    c++;
                }
            }
        }
        System.out.println(c);
    }

    private static int sumOfSquaredDifferences(int[] a, int[] b) {
        int sum = 0;
        for (int i = 0; i < a.length; i++) {
            sum += Math.pow(a[i] - b[i], 2);
        }
        return sum;
    }
}

File: dataset/codenet/Python/Code/s663509070.py
N,D=map(int,input().split())
mat=[]
for d in range(N):
     mat.append([int(x) for x in input().split()])

import math
c=0
for i in range(len(mat)-1):
    for j in range(i+1,len(mat)):
        dist=math.sqrt(sum([(x-y)**2 for x,y in zip(mat[i],mat[j])]))
        if int(dist)==dist:
                c+=1
print(c)